inc/Engine/Optimization/RUCSS/Warmup/ResourceFetcherProcess.php

```diff
@@ -0,0 +1,124 @@
+<?php
+declare(strict_types=1);
+
+namespace WP_Rocket\Engine\Optimization\RUCSS\Warmup;
+
+use WP_Rocket\Engine\Optimization\RUCSS\Database\Queries\ResourcesQuery;
+use \WP_Rocket_WP_Background_Process;
+
+class ResourceFetcherProcess extends WP_Rocket_WP_Background_Process {
+
+	/**
+	 * Background process action name.
+	 *
+	 * @var string
+	 */
+	protected $action = 'rucss_warmup_resource_fetcher';
+
+	/**
+	 * Background process prefix.
+	 *
+	 * @var string
+	 */
+
+	protected $prefix = 'rocket';
+
+	/**
+	 * ResourcesQuery instance.
+	 *
+	 * @var ResourcesQuery
+	 */
+	private $resources_query;
+
+	/**
+	 * APIClient instance.
+	 *
+	 * @var APIClient
+	 */
+	private $api_client;
+
+	/**
+	 * ResourceFetcherProcess constructor.
+	 *
+	 * @param ResourcesQuery $resources_query ResourcesQuery instance.
+	 * @param APIClient      $api_client APIClient instance.
+	 */
+	public function __construct( ResourcesQuery $resources_query, APIClient $api_client ) {
+		parent::__construct();
+
+		$this->resources_query = $resources_query;
+		$this->api_client      = $api_client;
+	}
+
+
+	/**
+	 * Do the task for each page resources.
+	 *
+	 * @param array $resource Resource array consists of url, content and type.
+	 *
+	 * @return false
+	 */
+	protected function task( $resource ) {
+		if ( ! is_array( $resource ) ) {
+			return false;
+		}
+
+		// check the database if those resources added before.
+		$db_row = $this->resources_query->get_item_by( 'url', $resource['url'] );
+
+		if ( empty( $db_row ) ) {
+			// Create this new row in DB.
+			$resource_id = $this->resources_query->add_item(
+				[
+					'url'           => $resource['url'],
+					'type'          => $resource['type'],
+					'content'       => $resource['content'],
+					'hash'          => md5( $resource['content'] ),
+					'last_accessed' => gmdate( 'Y-m-d\TH:i:s\Z' ),
```

Comment:
In Phase 3 I am using:
https://github.com/wp-media/wp-rocket/blob/65ad8426541fbd739133323aab4e434be4c0332b/inc/Engine/Optimization/RUCSS/Frontend/Subscriber.php#L109
For all datetime DB values. 
We shall uniform all the code to use a single datetime format.

https://developer.wordpress.org/reference/functions/current_time/

What do you think?

Reply:
We agreed on using modified and last_accessed only and use the function `current_time( 'mysql', true )` to use current GMT time.